include/zephyr/fs/ext2.h

```diff
@@ -27,7 +29,7 @@ struct ext2_cfg {
 	uint32_t block_size;
 	uint32_t fs_size; /* Number of blocks that we want to take. */
 	uint32_t bytes_per_inode;
-	uint8_t uuid[16];
+	struct uuid uuid;
```

Comment:
This is an API change, isn't it? 

If it is a stable API, then it needs to follow https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#introducing-breaking-api-changes

An alternative which is a bit dirty would be to do
```suggestion
	union {
		uint8_t uuid[16];
		struct uuid uuid;
	};
```

So that it's not breaking. I'll leave this up to the maintainer(s) of the `fs`

Reply:
It's true, I'm not sure how much the uuid field is used in the current version of the ext2 module. It would be nice if the maintainers of the `fs` could help.
Another option for this change to be non-breaking would be to keep the `typedef uint8_t uuid_t[UUID_SIZE];` in uuid.h instead of the struct version.

Reply:
@sorru94 I still think this needs to be considered an API change, which we probably should avoid or follow the procedure for breaking a stable API